cmd/sprint-automation/main.go

```diff
@@ -0,0 +1,195 @@
+	var o options
+	fs.StringVar(&o.logLevel, "log-level", "info", "Level at which to log output.")
+
+	for _, group := range []flagutil.OptionGroup{&o.jiraOptions} {
+		group.AddFlags(fs)
+	}
+
+	fs.StringVar(&o.slackTokenPath, "slack-token-path", "", "Path to the file containing the Slack token to use.")
+
+	if err := fs.Parse(args); err != nil {
+		logrus.WithError(err).Fatal("Could not parse args.")
+	}
+	return o
+}
+
+func main() {
+	logrusutil.ComponentInit()
+
+	o := gatherOptions(flag.NewFlagSet(os.Args[0], flag.ExitOnError), os.Args[1:]...)
+	if err := o.Validate(); err != nil {
+		logrus.WithError(err).Fatal("Invalid options")
+	}
+	level, _ := logrus.ParseLevel(o.logLevel)
+	logrus.SetLevel(level)
+
+	secretAgent := &secret.Agent{}
+	if err := secretAgent.Start([]string{o.slackTokenPath}); err != nil {
+		logrus.WithError(err).Fatal("Error starting secrets agent.")
+	}
+
+	prowJiraClient, err := o.jiraOptions.Client(secretAgent)
+	if err != nil {
+		logrus.WithError(err).Fatal("Could not initialize Jira client.")
+	}
+	jiraClient := prowJiraClient.JiraClient()
+	slackClient := slack.New(string(secretAgent.GetSecret(o.slackTokenPath)))
+
+	if err := postIssuesNeedingApproval(slackClient, jiraClient); err != nil {
+		logrus.WithError(err).Fatal("Could not post issues needing approval.")
+	}
+}
+
+const dptpTeamChannel = "team-dp-testplatform"
+
+func postIssuesNeedingApproval(slackClient *slack.Client, jiraClient *jiraapi.Client) error {
+	issues, response, err := jiraClient.Issue.Search(fmt.Sprintf(`project=%s AND status="QE Review"`, jira.ProjectDPTP), nil)
+	if err := jirautil.JiraError(response, err); err != nil {
+		return fmt.Errorf("could not find Jira project %s: %w", jira.ProjectDPTP, err)
+	}
+
+	if len(issues) == 0 {
+		return nil
+	}
+
+	blocks := []slack.Block{
+		&slack.HeaderBlock{
+			Type: slack.MBTHeader,
+			Text: &slack.TextBlockObject{
+				Type: slack.PlainTextType,
+				Text: "Cards Awaiting Acceptance",
+			},
+		},
+		&slack.SectionBlock{
+			Type: slack.MBTSection,
+			Text: &slack.TextBlockObject{
+				Type: slack.PlainTextType,
+				Text: "The following issues are ready for acceptance on the DPTP board:",
+			},
+		},
+	}
+	idByUser := map[string]slack.Block{}
+	blocksByUser := map[string][]slack.Block{}
+	for _, issue := range issues {
+		if _, recorded := idByUser[issue.Fields.Assignee.DisplayName]; !recorded {
+			idByUser[issue.Fields.Assignee.DisplayName] = &slack.ContextBlock{
+				Type: slack.MBTContext,
+				ContextElements: slack.ContextElements{
+					Elements: []slack.MixedElement{
+						&slack.ImageBlockElement{
+							Type:     slack.METImage,
+							ImageURL: issue.Fields.Assignee.AvatarUrls.Four8X48,
+							AltText:  issue.Fields.Assignee.DisplayName,
+						},
+						&slack.TextBlockObject{
+							Type: slack.MarkdownType,
+							Text: issue.Fields.Assignee.DisplayName,
+						},
+					},
+				},
+			}
+		}
+		// we really don't want these things to line wrap, so truncate the summary
```

Comment:
Why? Does slack has any limit or we don't want to create big messages?

Reply:
Slack has a limit to the width of an app's post, so if you use the full title it just line-wraps and makes the thing hard to read